### PR TITLE
Add start_index argument to all position embedding layers

### PR DESCRIPTION
**Pull request by mattdangerw**
This will allow all the layers to be used in a decoding context.

### CODE REVIEW

**Review by ianstenbit — APPROVED** [review]
LGTM -- might be worth some docstrings though!

File: keras_nlp/layers/modeling/rotary_embedding.py [hunk 1]
```diff
@@ -74,17 +74,19 @@ def __init__(
         self.feature_axis = feature_axis
         self.scaling_factor = scaling_factor
 
-    def call(self, inputs):
+    def call(self, inputs, start_index=0):
```
[the remark this answers is not in the document]
Comment (mattdangerw) [reply keras_nlp/layers/modeling/rotary_embedding.py:77]: Heh, yeah good call. Let me add that!

**Pull request merged**